celery_progress/websockets/backend.py

```diff
@@ -1,39 +1,32 @@
-import logging
-
 from celery_progress.backend import ProgressRecorder, Progress
 
 try:
     from asgiref.sync import async_to_sync
     from channels.layers import get_channel_layer
 except ImportError:
-    async_to_sync = get_channel_layer = None
-    WEBSOCKETS_AVAILABLE = False
+    channel_layer = None
 else:
-    WEBSOCKETS_AVAILABLE = get_channel_layer()
-
+    channel_layer = get_channel_layer()
 
-logger = logging.getLogger(__name__)
+if not channel_layer:
+    RuntimeError(
+        'Tried to use websocket progress bar, but dependencies were not installed / configured. '
+        'Use pip install celery-progress[websockets] and set up channels to enable this feature. '
+        'See: https://channels.readthedocs.io/en/latest/ for more details.'
+    )
 
 
 class WebSocketProgressRecorder(ProgressRecorder):
 
     @staticmethod
     def push_update(task_id):
-        if WEBSOCKETS_AVAILABLE:
-            try:
-                channel_layer = get_channel_layer()
-                async_to_sync(channel_layer.group_send)(
-                    task_id,
-                    {'type': 'update_task_progress', 'data': {**Progress(task_id).get_info()}}
-                )
-            except AttributeError:  # No channel layer to send to, so ignore it
-                pass
-        else:
-            logger.info(
-                'Tried to use websocket progress bar, but dependencies were not installed / configured. '
-                'Use pip install celery-progress[websockets] and setup channels to enable this feature.'
-                'See: https://channels.readthedocs.io/en/latest/ for more details.'
+        try:
+            async_to_sync(channel_layer.group_send)(
+                task_id,
+                {'type': 'update_task_progress', 'data': {**Progress(task_id).get_info()}}
             )
+        except AttributeError:  # No channel layer to send to, so ignore it
+            pass
```

Comment:
is this path reachable anymore?

Reply:
If the default channel layer cannot be found, and due to the lack of properly raising the error as explained above, the AttributeError exception will still be triggered by the post-run handler. 

One thing I've been thinking of is that currently, the nature of `get_channel_layer()` just looks for a channel layer named "default", and if none can be found, returns None instead. Should we offer the ability to configure the name of this layer, so that other layers are not potentially impacted?

Reply:
I've never actually used the websocket implementation - so it's not a hugely informed opinion - but at a high level configuring it by name sounds useful.